src/FSharp.FGL.IO/GDF.fs

```diff
@@ -136,3 +136,84 @@ module GDF =
         path
         |> System.IO.File.ReadAllLines
         |> fromArray
+
+
+    //Returns the GDFValue-type as string
+    let typeOfGDFValue (value:GDFValue) =    
+        match value with
+            | VARCHAR     x   -> "VARCHAR"
+            | BOOLEAN     x   -> "BOOLEAN"
+            | DOUBLE      x   -> "DOUBLE"
+            | INT         x   -> "INT"      
+
+    //Applies typeOfGDFValue on all GDF Values of a given Map
+    let private getGDFValue (value:Map<'key,GDFValue>) =
+        let valueList = Map.toList value|>List.map(snd)
+        List.map(typeOfGDFValue) valueList
+
+    //Adds quotationmarks at the beginning and the end of strings
+    let private addQuotationsToString (x:string) : string =
+        let sb = StringBuilder()
+        let charList = x.ToCharArray()|>List.ofArray
+        let updatedCharList = [''']@charList@[''']
+        for i in updatedCharList do sb.Append(i)
+        sb.ToString()
+
+    //Checks if a string contains commas, if true then applies addQuotationsToString
+    let private addQuotationsIfNeeded (x:string) :string =
+        if x.Contains "'" then addQuotationsToString x
+        else x
+
+    //Reconstructs the header of an edge or node based on a Map<string,GDFValue> 
+    let private reconstructHeader (value:Map<string,GDFValue>) =
+        let headerString    =   value|>Map.toList|>List.map(fst)
+        let headerTypes     =   value|>getGDFValue 
+        let header          =   List.map2 (fun a b -> (a)+" "+(b)) headerString headerTypes
+        let defIndex        =   header|>List.findIndex(fun x -> x.Contains "def>")
+        let headerSortedNew =   ((header.[defIndex])::header)|>List.distinct
+        String.concat ";" headerSortedNew
```

Comment:
You shouldn't concat your strings here, as you lose the ability to check for differences in entries between your vertices/edges